hibernate-core/src/main/java/org/hibernate/engine/jdbc/env/internal/JdbcEnvironmentInitiator.java

```diff
@@ -73,105 +71,63 @@ public JdbcEnvironment initiateService(Map<String, Object> configurationValues,
 								AvailableSettings.JAKARTA_HBM2DDL_DB_NAME
 						);
 					}
-					return value;
+					return (String) value;
+				}
+		);
+
+		// todo check this after major/minor version, and set only if none of those are set?
```

Comment:
Assuming you mean we move this evaluation below the major/minor ones, I think you mean as first argument, otherwise the additional supplier will not be considered?

Reply:
Yes, move below major/minor ones and write this:

```java
String explicitDatabaseVersion = NullnessHelper.coalesceSuppliedValues(
		() -> (String) configurationValues.get( AvailableSettings.JAKARTA_HBM2DDL_DB_VERSION ),
		() -> {
			final Object value = configurationValues.get( AvailableSettings.DIALECT_DB_VERSION );
			if ( value != null ) {
				DeprecationLogger.DEPRECATION_LOGGER.deprecatedSetting(
						AvailableSettings.DIALECT_DB_VERSION,
						AvailableSettings.JAKARTA_HBM2DDL_DB_VERSION
				);
			}
			return (String) value;
		},
		() -> {
			if ( explicitDatabaseMajorVersion != null ) {
				return explicitDatabaseMinorVersion == null ? explicitDatabaseMajorVersion.toString() : (explicitDatabaseMajorVersion.toString() + "." explicitDatabaseMinorVersion.toString());
			}
			return null;
		}
);
```

Reply:
Also, please add the following code after that code block I posted:

```java
if ( explicitDatabaseMajorVersion == null && explicitDatabaseMinorVersion == null && explicitDatabaseVersion != null ) {
	final String[] parts = explicitDatabaseVersion.split( "\\." );
	try {
		final int potentialMajor = Integer.parseInt( parts[0] );
		if ( parts.length > 1 ) {
			explicitDatabaseMinorVersion = Integer.parseInt( parts[1] );
		}
		explicitDatabaseMajorVersion = potentialMajor;
	}
	catch (NumberFormatException e) {
		// Ignore
	}
}
```